rust/gui-client/src-tauri/src/client.rs

```diff
@@ -109,15 +109,33 @@ fn run_gui(cli: Cli) -> Result<()> {
         logger: _logger,
         reloader,
     } = start_logging(&settings.log_filter)?;
-    let result = gui::run(cli, settings, reloader, telemetry);
 
-    // Make sure errors get logged, at least to stderr
-    if let Err(error) = &result {
-        tracing::error!(error = std_dyn_err(error), error_msg = %error);
-        common::errors::show_error_dialog(error.user_friendly_msg())?;
-    }
+    match gui::run(cli, settings, reloader, telemetry) {
+        Ok(()) => Ok(()),
+        Err(anyhow) => {
+            if anyhow
+                .chain()
+                .find_map(|e| e.downcast_ref::<tauri_runtime::Error>())
+                .is_some_and(|e| matches!(e, tauri_runtime::Error::CreateWebview(_)))
+            {
+                common::errors::show_error_dialog("Firezone cannot start because WebView2 is not installed. Follow the instructions at <https://www.firezone.dev/kb/client-apps/windows-client>.".to_string())?;
+                return Err(anyhow);
+            }
+
+            if anyhow.root_cause().is::<deep_link::CantListen>() {
+                common::errors::show_error_dialog(
+                    "Firezone is already running. If it's not responding, force-stop it."
+                        .to_string(),
+                )?;
+                return Err(anyhow);
+            }
 
-    Ok(result?)
+            common::errors::show_error_dialog(anyhow.to_string())?;
```

Comment:
For now I think this is ok, as long as we're on the quest to make the GUI client more robust.

Just want to note here that each time a user complains to their admin, we run a non-negligible risk of losing that customer. By the time the admin reaches out in our support channel they're already annoyed, as are a handful of users. So long-term any extra effort we invest into making errors as user-actionable as possible so they don't have to bug their admins for a fix will pay off (as I learned the hard way with the macOS client and system extension...)